File: lib/rgfa/line/group/path/to_gfa2.rb
module RGFA::Line::Group::Path::ToGFA2

  def to_gfa2_a
    items = []
    captured_path.each do |oline|
      case oline.line
      when RGFA::Line::Segment::GFA1
        items << oline.to_s
      when RGFA::Line::Edge::Link
        eid = oline.line.eid
        if eid.placeholder?
          raise RGFA::ValueError,
            "Links has no identifier\n"+
            "Path conversion to GFA2 failed"
        end
        items << eid + oline.orient.to_s
      end
    end
    a = ["O"]
    a << field_to_s(:path_name)
    a << items.join(" ")
    return a
  end

end

File: gfapy/line/group/path/to_gfa2.py
import gfapy

class ToGFA2:

  def _to_gfa2_a(self):
    items = []
    for oline in self.captured_path:
      if isinstance(oline.line, gfapy.line.segment.GFA1):
        items.append(str(oline))
      elif isinstance(oline.line, gfapy.line.edge.Link):
        items.append(oline.line.eid + str(oline.orient))
    a = ["O"]
    a.append(self.field_to_s("path_name"))
    a.append(" ".join(items))
    for tn in self.tagnames:
      a.append(self.field_to_s(tn, tag=True))
    return a
